test/rubygems/test_gem_commands_owner_command.rb

```diff
@@ -425,6 +425,42 @@ def test_with_webauthn_enabled_failure
     refute_match response_success, @stub_ui.output
   end
 
+  def test_with_webauthn_enabled_with_safari_as_default_browser
+    omit("Safari is only available on macOS") unless RUBY_PLATFORM.include?("darwin")
+
+    set_default_browser_safari
+    webauthn_verification_url = "rubygems.org/api/v1/webauthn_verification/odow34b93t6aPCdY"
+    response_fail = "You have enabled multifactor authentication but your request doesn't have the correct OTP code. Please check it and retry."
+    response_success = "Owner added successfully."
+    port = 5678
+    server = TCPServer.new(port)
+
+    @stub_fetcher.data["#{Gem.host}/api/v1/webauthn_verification"] = HTTPResponseFactory.create(body: webauthn_verification_url, code: 200, msg: "OK")
+    @stub_fetcher.data["#{Gem.host}/api/v1/gems/freewill/owners"] = [
+      HTTPResponseFactory.create(body: response_fail, code: 401, msg: "Unauthorized"),
+      HTTPResponseFactory.create(body: response_success, code: 200, msg: "OK"),
+    ]
+
+    TCPServer.stub(:new, server) do
+      Gem::WebauthnListener.stub(:wait_for_otp_code, "Uvh6T57tkWuUnWYo") do
+        use_ui @stub_ui do
+          @cmd.add_owners("freewill", ["user-new1@example.com"])
+        end
+      end
+    ensure
+      server.close
+    end
+
+    restore_default_browser
```

Comment:
Should we restore default browser in `ensure` block? If exception was happend after `set_default_browser_safari`, developer configuration is no longer restored.

Reply:
Ah good catch, will do!